assets/src/components/activities/oli_embedded/OliEmbeddedAuthoring.tsx

```diff
@@ -0,0 +1,252 @@
+import { configureStore } from 'state/store';
+import React from 'react';
+import {
+  AuthoringElement,
+  AuthoringElementProps,
+  AuthoringElementProvider,
+  useAuthoringElementContext,
+} from 'components/activities/AuthoringElement';
+import { OliEmbeddedModelSchema } from 'components/activities/oli_embedded/schema';
+import ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
+import * as ActivityTypes from 'components/activities/types';
+import { OliEmbeddedActions } from 'components/activities/oli_embedded/actions';
+
+import guid from 'utils/guid';
+import { uploadFiles } from 'components/media/manager/upload';
+import { CloseButton } from 'components/misc/CloseButton';
+import { MediaItemRequest, ScoringStrategy } from 'components/activities/types';
+import { lastPart } from 'components/activities/oli_embedded/utils';
+import { XmlEditor } from 'components/common/XmlEditor';
+import ModalSelection from 'components/modal/ModalSelection';
+const store = configureStore();
+
+const Embedded = (props: AuthoringElementProps<OliEmbeddedModelSchema>) => {
+  const { dispatch, model, onRequestMedia } = useAuthoringElementContext<OliEmbeddedModelSchema>();
+
+  const { projectSlug } = props;
+
+  function select(_projectSlug: string): Promise<string> {
+    return new Promise((resolve, reject) => {
+      const request = {
+        type: 'MediaItemRequest',
+      } as MediaItemRequest;
+      if (props.onRequestMedia) {
+        props.onRequestMedia(request).then((r) => {
+          if (r === false) {
+            reject('error');
+          } else {
+            resolve(r as string);
+          }
+        });
+      } else {
+        reject('error');
+      }
+    });
+  }
+
+  const addFile = (e: any) => {
+    select(projectSlug).then((url: string) => {
+      dispatch(OliEmbeddedActions.addResourceURL(url));
+    });
+  };
+
+  const display = (c: any, id: string) => {
+    let element = document.querySelector(id);
+    if (!element) {
+      element = document.createElement('div');
+      element.id = id;
```

Comment:
still not sure you are understanding me, if you assign `element.id = id;` but pass in "#upload_error" as id then that is fine for `document.querySelector('#upload_error');` but if that isn't found then I don't think you want the element's ID to be `#upload_error` but rather just `upload_error`

Reply:
I see what you mean. Updated with correction